processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt

```diff
@@ -66,6 +67,14 @@ class DataCompatProcessor(
             val classKdoc = classDeclaration.docString
             val packageName = classDeclaration.packageName.asString()
 
+            val otherAnnotations = classDeclaration.annotations
+                .filter { it.annotationType.resolve().toString() != DataCompat::class.simpleName }
+            // TODO tbh we actually should support only non-parametrized interfaces,
+            //  but checking just if KSClassDeclaration for simplicity
+            val implementedInterfaces = classDeclaration
+                .superTypes
+                .filter { it.resolve().declaration is KSClassDeclaration }
```

Comment:
Does this mean only kotlin interfaces will be kept?

Reply:
Kotlin classes, interfaces and objects. ToDo states that ideally we should limit ourselves just to non-parametrized interfaces but did not find an easy way for that

Reply:
appeared to be relatively easy